code/model/modelread.cpp

```diff
@@ -4972,9 +4970,70 @@ void swap_bsp_tmappoly( polymodel * pm, ubyte * p )
 	}
 }
 
+void swap_bsp_tmap2poly(polymodel* pm, ubyte* p)
+{
+	uint i, nv;
+	model_tmap_vert* verts;
+	vec3d* normal = vp(p + 8); // tigital
+	vec3d* center = vp(p + 20);
+	float radius = INTEL_FLOAT(&fl(p + 32));
+
+	fl(p + 32) = radius;
+
+	normal->xyz.x = INTEL_FLOAT(&normal->xyz.x);
+	normal->xyz.y = INTEL_FLOAT(&normal->xyz.y);
+	normal->xyz.z = INTEL_FLOAT(&normal->xyz.z);
+	center->xyz.x = INTEL_FLOAT(&center->xyz.x);
+	center->xyz.y = INTEL_FLOAT(&center->xyz.y);
+	center->xyz.z = INTEL_FLOAT(&center->xyz.z);
+
+	nv = INTEL_INT(uw(p + 36)); // tigital
+	uw(p + 36) = nv;
+
+	int tmap_num = INTEL_INT(w(p + 40)); // tigital
```

Comment:
Either declare variables as you need them, or declare them C style, not a mixture within the same function.